Fix ArchUnit tests

ArchUnit tests in the `store` module are failing with the following error:
```
org.junit.platform.commons.JUnitException: TestEngine with ID 'archunit' failed to discover tests
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:160)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:134)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:80)
	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:110)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: com.tngtech.archunit.junit.ArchTestInitializationException: java.lang.reflect.InvocationTargetException
	at com.tngtech.archunit.junit.ArchTestInitializationException$1.apply(ArchTestInitializationException.java:27)
	at com.tngtech.archunit.junit.ArchTestInitializationException$1.apply(ArchTestInitializationException.java:24)
	at com.tngtech.archunit.junit.ReflectionUtils.getValueOrThrowException(ReflectionUtils.java:102)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.getValue(ArchUnitTestDescriptor.java:111)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.resolveChildren(ArchUnitTestDescriptor.java:106)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.access$000(ArchUnitTestDescriptor.java:42)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor$ArchUnitArchTestsDescriptor.lambda$createChildren$0(ArchUnitTestDescriptor.java:216)
	at com.tngtech.archunit.junit.ElementResolver.handleNewSegment(ElementResolver.java:125)
	at com.tngtech.archunit.junit.ElementResolver.resolve(ElementResolver.java:79)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor$ArchUnitArchTestsDescriptor.lambda$createChildren$1(ArchUnitTestDescriptor.java:215)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor$DeclaredArchTests.handleFields(ArchUnitTestDescriptor.java:245)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor$ArchUnitArchTestsDescriptor.createChildren(ArchUnitTestDescriptor.java:214)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.lambda$resolveArchRules$6(ArchUnitTestDescriptor.java:124)
	at com.tngtech.archunit.junit.ElementResolver$RequestedButUnresolvedClass.ifRequestedButUnresolved(ElementResolver.java:201)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.resolveArchRules(ArchUnitTestDescriptor.java:121)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.resolveChildren(ArchUnitTestDescriptor.java:104)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.lambda$resolveField$4(ArchUnitTestDescriptor.java:92)
	at com.tngtech.archunit.junit.ElementResolver.handleNewSegment(ElementResolver.java:125)
	at com.tngtech.archunit.junit.ElementResolver.resolve(ElementResolver.java:79)
	at com.tngtech.archunit.junit.ElementResolver$UnresolvedMember.ifUnresolved(ElementResolver.java:229)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.resolveField(ArchUnitTestDescriptor.java:92)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.lambda$createChildren$2(ArchUnitTestDescriptor.java:85)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.createChildren(ArchUnitTestDescriptor.java:85)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.createTestDescriptor(ArchUnitTestDescriptor.java:77)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.lambda$resolve$0(ArchUnitTestDescriptor.java:62)
	at com.tngtech.archunit.junit.ElementResolver$RequestedButUnresolvedClass.ifRequestedButUnresolved(ElementResolver.java:201)
	at com.tngtech.archunit.junit.ArchUnitTestDescriptor.resolve(ArchUnitTestDescriptor.java:62)
	at com.tngtech.archunit.junit.ArchUnitTestEngine.lambda$resolveRequestedClasses$3(ArchUnitTestEngine.java:118)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at com.tngtech.archunit.junit.ArchUnitTestEngine.resolveRequestedClasses(ArchUnitTestEngine.java:118)
	at com.tngtech.archunit.junit.ArchUnitTestEngine.discover(ArchUnitTestEngine.java:80)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:152)
	... 13 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.tngtech.archunit.base.ReflectionUtils.newInstanceOf(ReflectionUtils.java:28)
	at com.tngtech.archunit.junit.ReflectionUtils.newInstanceOf(ReflectionUtils.java:81)
	at com.tngtech.archunit.junit.ReflectionUtils.getValueOrThrowException(ReflectionUtils.java:99)
	... 53 more
Caused by: java.lang.NoClassDefFoundError: org/springframework/data/jpa/repository/Modifying
	at com.backbase.buildingblocks.archunit.RelationalPersistenceRules.<clinit>(RelationalPersistenceRules.java:40)
	at com.backbase.buildingblocks.archunit.test.AllRelationalPersistenceRules.<init>(AllRelationalPersistenceRules.java:12)
	... 60 more
Caused by: java.lang.ClassNotFoundException: org.springframework.data.jpa.repository.Modifying
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 62 more

Process finished with exit code -2
```

After the initial investigation, it seems that the root cause is the following rule:
```java
    @ArchTest
    ArchTests relationalPersistenceRules = ArchTests.in(AllRelationalPersistenceRules.class);
```
Which requires some classes from the Spring Data JPA. 

These tests start failing after the following dependency was removed. 
```xml
        <dependency>
            <groupId>org.springframework.data</groupId>
            <artifactId>spring-data-jpa</artifactId>
            <scope>test</scope>
        </dependency>
```
See commit - https://github.com/Backbase/golden-sample-services/commit/206d5464b1f225747dfd6daf4ee8d35f72171cd8


Acceptance criteria:
- ArchUnit tests can be executed (from CLI and IDE)
- Fix failing tests (if any)

Solution direction:
- Add missing dependency back to the `pom.xml`

--- a/services/store/src/test/java/com/backbase/goldensample/store/archunit/ArchUnitRulesTest.java
+++ b/services/store/src/test/java/com/backbase/goldensample/store/archunit/ArchUnitRulesTest.java
@@ -1,19 +1,16 @@
 package com.backbase.goldensample.store.archunit;
 
-import com.backbase.buildingblocks.archunit.ArchitectureRules;
 import com.backbase.buildingblocks.archunit.test.AllArchitectureRules;
 import com.backbase.buildingblocks.archunit.test.AllConfigurationRules;
 import com.backbase.buildingblocks.archunit.test.AllControllerRules;
 import com.backbase.buildingblocks.archunit.test.AllGeneralCodingRules;
 import com.backbase.buildingblocks.archunit.test.AllLoggingRules;
 import com.backbase.buildingblocks.archunit.test.AllNamingConventionRules;
-import com.backbase.buildingblocks.archunit.test.AllRelationalPersistenceRules;
 import com.backbase.goldensample.store.Application;
 import com.tngtech.archunit.core.importer.ImportOption;
 import com.tngtech.archunit.junit.AnalyzeClasses;
 import com.tngtech.archunit.junit.ArchTest;
 import com.tngtech.archunit.junit.ArchTests;
-import com.tngtech.archunit.lang.syntax.elements.GivenClassesConjunction;
 
 @AnalyzeClasses(packagesOf = Application.class, importOptions = ImportOption.DoNotIncludeTests.class)
 public class ArchUnitRulesTest {
@@ -35,8 +32,4 @@ public class ArchUnitRulesTest {
 
     @ArchTest
     ArchTests namingConventionRules = ArchTests.in(AllNamingConventionRules.class);
-
-    // No Persistence in this module.
-    // @ArchTest
-    // ArchTests relationalPersistenceRules = ArchTests.in(AllRelationalPersistenceRules.class);
 }

